app/models/risk_condition.rb

```diff
@@ -11,44 +11,107 @@ class RiskCondition < ApplicationRecord
   # :no_submissions => :no_submissions_threshold
   # :low_grades => :grade_threshold, :count_threshold
 
-  def self.create_condition_for_course_with_type(course_id, type, params)
+  def self.create_condition_for_course_with_type(course_id, type, params, version)
   	# parameter check shouldn't surface to user and is for sanity check during development only
-    if (type == :grace_day_usage && (params[:grace_day_threshold].nil? || params[:date].nil? || params.length != 2)) ||
-       (type == :grade_drop && (params[:percentage_drop].nil? || params[:consecutive_counts].nil? || params.length != 2)) ||
-       (type == :no_submissions && (params[:no_submissions_threshold].nil? || params.length != 1)) ||
-       (type == :low_grades && (params[:grade_threshold].nil? || params[:count_threshold].nil? || params.length != 2))
+    if (type == 1 && (params[:grace_day_threshold].nil? || params[:date].nil? || params.length != 2)) ||
+       (type == 2 && (params[:percentage_drop].nil? || params[:consecutive_counts].nil? || params.length != 2)) ||
+       (type == 3 && (params[:no_submissions_threshold].nil? || params.length != 1)) ||
+       (type == 4 && (params[:grade_threshold].nil? || params[:count_threshold].nil? || params.length != 2))
       raise "Parameters for risk condition does not match type!"
     end
 
-    version = RiskCondition.get_version(course_id, type)
     options = { course_id: course_id, condition_type: type, parameters: params, version: version }
     newRiskCondition = RiskCondition.new(options)
```

Comment:
Nit picking here. but you were using underscore casing, but this one is in camel case. Suggest changing to new_risk_condition to be consistent

Reply:
See latest commit.